### PR TITLE
Use CreateFileW on Windows to create file in local_drive_create_file()

### LINKED ISSUE #3745: Mortal Kombat 1+2 CD config saving issue
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

0.81.1

### What Operating System are you using?

Windows 11

### If Other OS, please describe

_No response_

### Relevant hardware info

Ryzen 3700x, Nvidia RTX 4070ti 

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

Saving the options in Mortal Kombat 1 or 2 doesn't work after the first time they're set.

I broadly know what the cause of the problem is: depending on the game, the options are saved to the file CMOS.MK1 or CMOS.MK2. The problem seems to be that the game sets the attributes of the options file to hidden and system, and Dosbox seems to no longer be able to write to the file once those are set.

### Steps to reproduce the behaviour.

Explain how to reproduce
1. Start the game
2. Press F10 to get to the options screen
3. Set options and exit.
4. Go back to the options and change something.
5. Exit the game.
6. Restart the game.
7. The options last set will not have stuck.

### Download URL of affected game or software

_No response_

### Your configuration

_No response_

### Provide a Log

_No response_

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

### PR DESCRIPTION
**Pull request by weirddan455**
# Description

This reverts commit ff90cc5666e2be79e62afe624634c826e51c33f3.

Additionally, add file sharing modes FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE to fix the Crystal Caves save issue.

## Related issues

Fixes #3745

#3262 (crystal caves issue)
#3265 (crystal caves PR)

# Manual testing

Crystal Caves save works.  I was able to reproduce the bug and adding sharing flags allows the file to be re-created even if another file handle is already open due to game leaks.

Mortal Kombat save works.  For some reason, `fopen` fails to open files marked with the system attribute.  Reverting back to `CreateFileW` works to solve this.

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [x] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

**Comment by johnnovak** [comment]
>  For some reason, fopen fails to open files marked with the system attribute. Reverting back to CreateFileW works to solve this.

Kinda/sorta makes sense, and maybe the C standard stipulates that or something. Usually, platform-specific APIs that interact with the host OS stuff are a lot better anyway than the C or POSIX stuff (at least on Windows). C/POSIX is often kinda a "poor man's fallback" on Windows 😆

File: src/misc/fs_utils_win32.cpp [hunk 1]
```diff
@@ -66,18 +66,23 @@ constexpr uint8_t WindowsAttributesMask = 0x3f;
...
-	if (file_pointer) {
-		if (!SetFileAttributesW(path.c_str(), attributes._data)) {
-			fclose(file_pointer);
-			return nullptr;
-		}
+	FILE* file_pointer          = nullptr;
+	const auto win32_attributes = (attributes._data != 0)
+	                                    ? static_cast<DWORD>(attributes._data)
+	                                    : FILE_ATTRIBUTE_NORMAL;
+	const auto win32_handle     = CreateFileW(path.c_str(),
```
Comment (johnnovak) [inline src/misc/fs_utils_win32.cpp:73]: picky: maybe insert a new line above this and reformat?

**Review by johnnovak — APPROVED** [review]
Nice quick fix @weirddan455 🏅

File: src/misc/fs_utils_win32.cpp (hunk 1 above)
Comment (weirddan455) [reply src/misc/fs_utils_win32.cpp]: Formatted

**Pull request merged**